internal/subscription/core.go

```diff
@@ -72,3 +75,20 @@ func (c *Core) Exists(ctx context.Context, key string) (bool, error) {
 	}
 	return ok, nil
 }
+
+// DeleteSubscription deletes a subscription
+func (c *Core) DeleteSubscription(ctx context.Context, m *Model) error {
```

Comment:
should we also validate if the subscription belongs to a project as well

Reply:
This is being checked here. https://github.com/razorpay/metro/pull/44/files#diff-62bfaba279adf117168ffb1d9690cc9c5d7227ad67cd181eb9934b4330bdf779R87 
key() returns the full consul path registry/projects/<project>/subscriptions/<subscription>